arches/app/models/graph.py

```diff
@@ -1586,15 +1589,18 @@ def publish(self, notes=None):
         Adds a row to the GraphPublication table
         Assigns GraphPublication id to Graph
         """
-        publication = models.GraphPublication.objects.create(
-            graph=self,
-            serialized_graph=JSONSerializer().serialize(self, force_recalculation=True),
-            notes=notes,
-        )
-        publication.save()
+        try:
+            publication = models.GraphPublication.objects.create(
+                graph=self,
+                serialized_graph=JSONDeserializer().deserialize(JSONSerializer().serialize(self, force_recalculation=True)),
+                notes=notes,
+            )
+            publication.save()
 
-        self.publication = publication
-        self.save()
+            self.publication = publication
+            self.save()
+        except Exception:
+            raise UnpublishedModelError("Could not save graph publication ")
```

Comment:
can this error be more descriptive, can we leverage the exception?

Reply:
not sure how much more descriptive it can be; happy to get your thoughts on this.

Reply:
oh sure, instead of that sentence can pass the declared Exception? `except Exception as e:`